src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -17,60 +18,77 @@ const Hoverboard = dynamic(() => import("./Hoverboard"), {
     setHoverboardColor(colorOptions[randomIndex]);
-  };
+  }, []);
 
   useEffect(() => {
-    const interval = setInterval(changeHoverboardColor, 5_000);
-    return () => clearInterval(interval);
-  }, [hoverboardColor]);
+    const timeoutId = setTimeout(changeHoverboardColor, 5000);
+    return () => clearTimeout(timeoutId);
+  }, [changeHoverboardColor]);
 
   return (
-    <BubbleViewportWrapper>
-      <div className="relative flex w-96 flex-col items-center space-y-1 rounded-lg bg-loadingBoxes p-4 py-12 shadow-sm">
-        <div className="flex flex-col items-center space-y-2">
-          <div className="h-32 w-32" onClick={changeHoverboardColor}>
-            <Hoverboard color={hoverboardColor} />
+    <div className={styles.loadingScreenWrapper}>
+      <BubbleViewportWrapper>
+        <div className={styles.loadingScreenWrapper}>
+          <div className="flex flex-col items-center space-y-2">
+            <div className={styles.hoverboardWrapper} onClick={changeHoverboardColor}>
+              <Hoverboard color={hoverboardColor} />
+            </div>
+            {children}
           </div>
-          {children}
         </div>
-      </div>
-    </BubbleViewportWrapper>
+      </BubbleViewportWrapper>
+    </div>
   );
 }
 
+const phrases = [
+  "Tuning the flux capacitor...",
+  "Prepping DeLorean for time travel...",
+  "Revving up to 88 miles per hour...",
+  "Gathering 1.21 gigawatts of power...",
+  "Ensuring temporal paradox safeguards...",
+  "Being careful not to mess up your timeline...",
+];
+
 function LoadingScreen({
   uploading,
   awaitingSourcemaps,
   fallbackMessage,
-  stalledTimeout = 9000,
+  stalledTimeout = 15000,
 }: PropsFromRedux & { fallbackMessage: string; stalledTimeout?: number }) {
-  // The backend send events in this order: uploading replay -> uploading sourcemaps.
-  let waitingForMessage = <span>{fallbackMessage}</span>;
-  if (awaitingSourcemaps) {
-    waitingForMessage = <span>Uploading sourcemaps...</span>;
-    stalledTimeout = Infinity;
-  } else if (uploading) {
-    waitingForMessage = <span>Uploading {Math.round(+uploading.amount)}Mb</span>;
-    stalledTimeout = Infinity;
-  }
-
-  let [stalled, setStalled] = useState<boolean>(false);
-  useEffect(() => {
-    if (stalledTimeout === Infinity) {
-      return;
-    }
+  const longWaitMessage =
+    "<div><p>This is taking longer than expected.</p><p><a href='http://replay.io/discord' target='discord'>Contact us on Discord</a></p>";
 
-    const timeout = setTimeout(() => setStalled(true), stalledTimeout);
+  const [message, setMessage] = useState(fallbackMessage);
 
-    return () => clearTimeout(timeout);
-  }, [fallbackMessage, stalledTimeout]);
+  useEffect(() => {
+    const changeMessage = () => {
+      const randomIndex = Math.floor(Math.random() * phrases.length);
+      setMessage(phrases[randomIndex]);
+    };
+    const phraseTimeout = setTimeout(changeMessage, 5000);
+    return () => clearTimeout(phraseTimeout);
```

Comment:
Since the effect will only run once now (good) this will only change the message once (maybe not what you wanted?)

Reply:
Ah yes, good call. This is indeed what is designed. We only want a single cutesy message to appear before the stalled message, so it'll go like this:

1. 5 seconds of "loading"
2. 10 seconds of a single cutesy message
3. Stalled message that never goes away